src/browser/BrowserService.cpp

```diff
@@ -925,60 +928,75 @@ int BrowserService::sortPriority(const Entry* entry,
                                  const QString& baseSubmitUrl,
                                  const QString& fullUrl) const
 {
-    QUrl url(entry->url());
-    if (url.scheme().isEmpty()) {
-        url.setScheme("https");
-    }
+    auto getPriority = [&](const QString& givenUrl) {
+        QUrl url(givenUrl);
+        if (url.scheme().isEmpty()) {
+            url.setScheme("https");
+        }
 
-    // Add the empty path to the URL if it's missing
-    if (url.path().isEmpty() && !url.hasFragment() && !url.hasQuery()) {
-        url.setPath("/");
-    }
+        // Add the empty path to the URL if it's missing
+        if (url.path().isEmpty() && !url.hasFragment() && !url.hasQuery()) {
```

Comment:
Wouldn't we want to add the empty path regardless of Fragment and Query also being present? Not sure how these matter or what the intent is here. At least a comment explaining it would be nice.

Reply:
Sure. I can add a comment. The path is appended because URL's passed from the extension always has a `/` as a path. Entry URL's doesn't require it, but we still want to match both cases.

Reply:
Yah but we are not adding the empty path if there is a fragment or query. For example: `https://example.com#anchor` would not add the empty path to the end.

Reply:
AFAIK those URL's will have the path already set.